e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,191 @@
+        new_master_pod_node = ''
+
+        body = {
+            "spec": {
+                "taints": [
+                    {
+                        "effect": "NoExecute",
+                        "key": "postgres"
+                    }
+                ]
+            }
+        }
+        podsList = k8s.core_v1.list_namespaced_pod("default", label_selector=labels)
+        for pod in podsList.items:
+            if ('spilo-role', 'master') in pod.metadata.labels.items():
+                master_pod_node = pod.spec.node_name
+            elif ('spilo-role', 'replica') in pod.metadata.labels.items():
+                new_master_pod_node = pod.spec.node_name
+
+        k8s.core_v1.patch_node(master_pod_node, body)
+        Utils.wait_for_master_failover(k8s, new_master_pod_node, self.RETRY_TIMEOUT_SEC)
+
+        self.assertTrue(master_pod_node != new_master_pod_node, "Master on {} did not fail over to {}".format(master_pod_node, new_master_pod_node))
+
+        # undo the tainting
+        body = {
+            "spec": {
+                "taints": []
+            }
+        }
+        k8s.core_v1.patch_node(new_master_pod_node, body)
+
+
+class K8sApi:
+
+    def __init__(self):
+
+        # https://github.com/kubernetes-client/python/issues/309
+        warnings.simplefilter("ignore", ResourceWarning)
+
+        self.config = config.load_kube_config()
+        self.k8s_client = client.ApiClient()
+        self.core_v1 = client.CoreV1Api()
+        self.crd = client.CustomObjectsApi()
+        self.apps_v1 = client.AppsV1Api()
+
+
+class Utils:
+
+    @staticmethod
+    def wait_for_pod_start(k8s_api, pod_labels, retry_timeout_sec):
+        pod_phase = 'No pod running'
+        while pod_phase != 'Running':
+            pods = k8s_api.core_v1.list_namespaced_pod('default', label_selector=pod_labels).items
+            if pods:
+                pod_phase = pods[0].status.phase
+        time.sleep(retry_timeout_sec)
+
+    @staticmethod
+    def wait_for_pg_to_scale(k8s_api, number_of_instances, retry_timeout_sec):
+
+        body = {
+            "spec": {
+                "numberOfInstances": number_of_instances
+            }
+        }
+        _ = k8s_api.crd.patch_namespaced_custom_object("acid.zalan.do",
+                                                       "v1", "default", "postgresqls", "acid-minimal-cluster", body)
+
+        labels = 'version=acid-minimal-cluster'
+        while Utils.count_pods_with_label(k8s_api, labels) != number_of_instances:
+            time.sleep(retry_timeout_sec)
+
+    @staticmethod
+    def count_pods_with_label(k8s_api, labels):
+        return len(k8s_api.core_v1.list_namespaced_pod('default', label_selector=labels).items)
+
+    @staticmethod
+    def wait_for_master_failover(k8s_api, expected_master_pod_node, retry_timeout_sec):
+        pod_phase = 'Failing over'
+        new_master_pod_node = ''
+        while (pod_phase != 'Running') & (new_master_pod_node != expected_master_pod_node):
```

Comment:
`&` is `bitwise and` ; the standard `and` keyword shall be preferred for readability purposes